### PR TITLE
Compiler warnings

### PR DESCRIPTION
**Pull request by bmiklautz**
* enable more compiler warnings
* fixes for the enabled warnings

### CODE REVIEW

**Comment by hardening** [comment]
+1

**Comment by akallabeth** [comment]
+1

File: rdtk/librdtk/rdtk_surface.h [hunk 1]
```diff
@@ -33,17 +33,5 @@ struct rdtk_surface
...
-#endif
-
-rdtkSurface* rdtk_surface_new(rdtkEngine* engine, BYTE* data, int width, int height, int scanline);
-void rdtk_surface_free(rdtkSurface* surface);
-
-#ifdef __cplusplus
-}
-#endif
-
 #endif /* RDTK_SURFACE_PRIVATE_H */
```
Comment (bmiklautz) [inline rdtk/librdtk/rdtk_surface.h]: Side note: these two functions are defined for export in rdtk/include/rdtk/rdtk.h .

**Comment by mfleisz** [comment]
+1

**Pull request merged**